cogs/Voting.py

```diff
@@ -0,0 +1,644 @@
+        self.title = title
+        self.DBManager = db_manager
+
+        self.target_roles = []
+        self.chair = None
+        self.target_voice_channel = None
+
+        self.options = []
+
+        self.sent_to = {}
+
+    def is_ready(self):
+        """
+        Check if the vote is ready to be sent out
+        :return: True if ready, False otherwise
+        """
+        return 1 < len(self.options) < 11
+
+    def add_role(self, role_id):
+        """
+        Adds a target role to send the vote to
+        :param role_id: target role
+        :return: None
+        """
+        self.target_roles.append(role_id)
+        self.DBManager.db_execute_commit("INSERT INTO vote_target_roles VALUES (?, ?)", (self.id, role_id))
+
+    def remove_role(self, role_id):
+        """
+        Removes target role from vote targets
+        :param role_id: target role
+        :return: None
+        """
+        self.target_roles.remove(role_id)
+        self.DBManager.db_execute_commit("DELETE FROM vote_target_roles WHERE vote_author_id=? AND role_id=?", (self.id, role_id))
+
+    def set_chair(self, chair_id=None):
+        """
+        Sets the chair of the vote to the given id
+        :param chair_id: target chair
+        :return: None
+        """
+        self.chair = chair_id
+        self.DBManager.db_execute_commit("UPDATE votes SET chair_id=? WHERE author_id=?", (chair_id, self.id))
+
+    def set_vc(self, channel_id=None):
+        """
+        Sets the target voice channel to a given channel id
+        :param channel_id: target discord voice channel id
+        :return: None
+        """
+        self.target_voice_channel = channel_id
+        self.DBManager.db_execute_commit("UPDATE votes SET voice_id=? WHERE author_id=?", (channel_id, self.id))
+
+    def add_option(self, option):
+        """
+        Adds an option to the vote
+        :param option: Option object
+        :return: None
+        """
+        self.options.append(option)
+        self.DBManager.db_execute_commit("INSERT INTO vote_options VALUES (?, ?, ?, ?)", (self.id, option.id, option.head, option.body))
+
+    def remove_option(self, index):
+        """
+        Removes an option from the vote
+        :param index: the location in the list of options to remove
+        :return: None
+        """
+        opt = self.options.pop(index-1)
+        self.DBManager.db_execute_commit("DELETE FROM vote_options WHERE vote_author_id=? AND opt_id=?", (self.id, opt.id))
+
+    def register_sent(self, user_id, msg_id):
+        """
+        Marks a user as having been sent a message to vote on
+        :param user_id: user who was sent the message
+        :param msg_id: the id of the message that was sent
+        :return:
+        """
+        self.sent_to[user_id] = msg_id
+        self.DBManager.db_execute_commit("INSERT INTO vote_sent VALUES (?, ?, ?)", (self.id, user_id, msg_id))
+
+
+emote_reference = TwoWay({0: "1️⃣", 1: "2️⃣", 2: "3️⃣",
```

Comment:
This should go at the top of the file 

```
# Futures

# Built-in/Generic Imports

# Libs

# Own modules

# Constants

# Variables
--- right here
```

Reply:
TwoWay is a class I've defined in the file so I can't put it at the top level and also before the definition of TwoWay. Could move the class to the utils cog though. It is just a quicker way for me to map emojis to specific numbers 1 to 1 be able to use keys as values/values as keys.

Reply:
nvm, have solved